src/main/scala/sbtassembly/Assembly.scala

```diff
@@ -347,8 +347,15 @@ object Assembly {
           .getOrElse(System.currentTimeMillis())
 
         timed(Level.Debug, "Create jar") {
-          IO.delete(output)
-          createJar(output, jarEntriesToWrite, jarManifest, localTime)
+          if (output.isDirectory) {
+            val invalidPath = output.toPath.toAbsolutePath.normalize
+            log.error(s"Attempted to overwrite existing directory: $invalidPath")
+            log.error("Update 'assemblyOutputPath' key or manually delete the corresponding path.")
```

Comment:
```suggestion
            log.error(s"expected a file name for assemblyOutputPath, but found a directory: ${invalidPath}; fix the setting or delete the directory")
```

"expected X but found Y" is probably clearer than "attempted to overwrite".